authentication/oauth_test.go

```diff
@@ -408,6 +408,58 @@ func TestOAuthWithIDTokenVerification(t *testing.T) {
 	})
 }
 
+func TestPushedAuthorizationRequest(t *testing.T) {
+	t.Run("Should require a client secret", func(t *testing.T) {
+		_, err := authAPI.OAuth.PushedAuthorization(context.Background(), oauth.PushedAuthorizationRequest{})
+		assert.ErrorContains(t, err, "client_secret or client_assertion is required but not provided")
+	})
+
+	t.Run("Should make a PAR request", func(t *testing.T) {
+		skipE2E(t)
+		configureHTTPTestRecordings(t, authAPI)
+
+		res, err := authAPI.OAuth.PushedAuthorization(context.Background(), oauth.PushedAuthorizationRequest{
+			ClientAuthentication: oauth.ClientAuthentication{
+				ClientSecret: clientSecret,
+			},
+			ResponseType: "code",
+			RedirectURI:  "http://localhost:3000/callback",
```

Comment:
Any way we can mark these as required?

Reply:
I'll add something to the function body to check for the required fields

Reply:
Added in 65a3389e78a4a1197c7b673f1dc32d155866e042